supreme/src/commonMain/kotlin/at/asitplus/signum/supreme/symmetric/discouraged/encrypt.kt

```diff
@@ -0,0 +1,94 @@
+package at.asitplus.signum.supreme.symmetric.discouraged
+
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import at.asitplus.signum.HazardousMaterials
+import at.asitplus.signum.indispensable.SecretExposure
+import at.asitplus.signum.indispensable.symmetric.*
+import at.asitplus.signum.indispensable.symmetric.SymmetricKey.WithDedicatedMac
+import at.asitplus.signum.supreme.symmetric.Encryptor
+import kotlin.jvm.JvmName
+
+
+/**
+ * Encrypts [data] using the manually specified [nonce]. Check yourself, before you really, really wreck yourself!
+ * * [nonce] =  _Initialization Vector_; **NEVER EVER RE-USE THIS!**
+ * * [authenticatedData] = _Additional Authenticated Data_
+ *
+ * It is safe to discard the reference to [nonce] and [authenticatedData], as both will be added to any  [SealedBox.Authenticated] resulting from an encryption.
+ *
+ * @return [KmmResult.success] containing a [SealedBox.Authenticated] if valid parameters were provided or [KmmResult.failure] in case of
+ * invalid parameters (e.g., key or nonce length)
+ */
+@HazardousMaterials("NEVER re-use a nonce/IV! Have them auto-generated instead!")
+@JvmName("encryptAuthenticatedWithNonce")
+suspend fun <K : KeyType, A : AuthCapability.Authenticated<out K>> KeyWithNonceAuthenticating<A, out K>.encrypt(
+    data: ByteArray,
+    authenticatedData: ByteArray? = null
+): KmmResult<SealedBox<A, NonceTrait.Required, out K>> = catching {
+    @OptIn(SecretExposure::class) Encryptor(
```

Comment:
also a file-level opt-in

Reply:
i took care to explicitly add it to every statement that requires it, and not file-level